kod/object/item/passitem/roomkeyc.kod

```diff
@@ -254,6 +254,16 @@ messages:
       propagate;
    }
 
+   Delete()
+   {
+      if (Send(self, @IsKeyCopy))
+      {
+         Send(&RoomKey, @RemoveCopy, #what=self);
```

Comment:
Sends to every key in the entire game?  Is there no backreference to the key this is a copy of?

Reply:
I originally tried that and it worked great when handing the key to an innkeeper to destroy, but when a room expires, it spits out errors--presumably because the room and associated gold key are also deleted, so the lookup chain that eventually finds the gold key ended up breaking in various ways.

While perhaps not ideal, this is the most foolproof way to make sure a copy gets unlinked from whatever gold key is storing a reference to it, no matter where it or the gold key are (or aren't) in the world.  To note, something similar is done with riija swords and sword shards when a player suicides.

I *can* try to find an alternative again if you like, especially if it results in server lag.